python/ray/data/_internal/execution/operators/zip_operator.py

```diff
@@ -236,11 +237,28 @@ def _zip_one_block(
         # Swap blocks if ordering was inverted during block alignment splitting.
         block, other_block = other_block, block
     # Zip block and other blocks.
-    result = BlockAccessor.for_block(block).zip(other_block)
+    result = _try_zip(BlockAccessor.for_block(block), other_block)
     br = BlockAccessor.for_block(result)
     return result, br.get_metadata(input_files=[], exec_stats=stats.build())
 
 
 def _get_num_rows_and_bytes(block: Block) -> Tuple[int, int]:
     block = BlockAccessor.for_block(block)
     return block.num_rows(), block.size_bytes()
+
+
+def _try_zip(block: BlockAccessor, other_block: Block) -> Block:
```

Comment:
nit: to keep things clean, can the parameter type be `Block` instead of `BlockAccessor`? also for the function name, can we indicate that we will try a fallback conversion to pyarrow?